.github/workflows/testing.yml

```diff
@@ -0,0 +1,159 @@
+          # Check codestyle
+          cd $MESHPY_PATH
+          black . --check --exclude="${PYTHON_VENV}" && exit 0
+          # If we did not exit earlier, raise an error here
+          exit 1
+  geometric-search-testing:
+    name: geometric-search-testing
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Additionally test the geometric search module with ArborX
+        run: |
+          whoami
+          pwd
+          MESHPY_PATH=$(pwd)
+          # Load the external modules
+          git submodule update --init
+          # Load the Kokkos module via spack
+          . ${SPACK_ACTIVATION_SCRIPT}
+          spack load kokkos@3.7.01
+          # Build the binaries
+          mkdir -p build/arborx
+          cd build/arborx
+          cmake ../../meshpy/geometric_search/src
+          make
+          cd $MESHPY_PATH
+          # Create the virtual environment
+          ${PYTHON_EXE} -m venv ${PYTHON_VENV}
+          source ${PYTHON_VENV}/bin/activate
+          # We have to update pip, otherwise there is a bug in the compilation of the cython code
+          pip install --upgrade pip
+          # Install meshpy, it is not yet possible to globally install the ArborX binaries, so we
+          # use the developement mode here.
+          pip install -e .
+          # Print information on the python environment
+          python --version
+          pip list
+          # Run tests
+          cd tests
+          python3 testing_geometric_search.py
+  performance-test:
+    name: meshpy-performance-testing
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Performace test
+        run: |
+          whoami
+          pwd
+          MESHPY_PATH=$(pwd)
+          # Create the virtual environment
+          ${PYTHON_EXE} -m venv ${PYTHON_VENV}
+          source ${PYTHON_VENV}/bin/activate
+          # We have to update pip, otherwise there is a bug in the compilation of the cython code
+          pip install --upgrade pip
+          # Install meshpy
+          pip install .
+          # Install cubitpy
+          cd $CUBITPY_REPO
+          git pull
+          pip install .
+          cd $MESHPY_PATH
+          # Print information on the python environment
+          python --version
+          pip list
+          # Run performance tests
+          cd tests
+          python3 performance_testing.py
+  deploy:
+    name: deploy
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: For the master branch, add the coverage report and the website
+        run: |
+          mkdir public
+          mkdir public/htmlcov
+          mv tests/htmlcov/ public/
+          cd website
+          bundle exec jekyll build -d public
+          cd public
+          mv * ../../public
+  
```

Comment:
This can be removed since it is related to the meshpy website. Maybe we can just create a similar website here on Github?